File: 125-valid-palindrome/valid-palindrome.java
class Solution {
    public boolean isPalindrome(String s) {
        int i = 0, j = s.length()-1;
        while(i < j){
            char tm1 = Character.toLowerCase(s.charAt(i));
            char tm2 = Character.toLowerCase(s.charAt(j));
            if(!Character.isLetterOrDigit(tm1)) i++;
            else if(!Character.isLetterOrDigit(tm2)) j--;
            else if(tm1 != tm2) return false;
            else{
                i++;
                j--;
            }
        }
        return true;
    }
}

File: 125-valid-palindrome/valid-palindrome.js
/**
 * @param {string} s
 * @return {boolean}
 */
var isPalindrome = function (s) {
    let i = 0, j = s.length - 1;
    let arr = s.split('');
    while (i < j) {
        let c1 = arr[i].toLowerCase();
        let c2 = arr[j].toLowerCase();
        if (!/[a-zA-Z0-9]/.test(c1)) {
            i++;
        } else if (!/[a-zA-Z0-9]/.test(c2)) {
            j--;
        } else if (c1 != c2) {
            return false;
        } else {
            i++;
            j--;
        }
    }

    return true;
};